### PR TITLE
Bug fix enhances april

### PR DESCRIPTION
**Pull request by Tynael**

### CODE REVIEW

File: ajax.php [hunk 1]
```diff
@@ -114,22 +146,22 @@ function getThumbnailAddToCartJS($id)
...
                     $js_variationDetails = implode(', ', $arr_variationDetails);
                     $js_variation = '{
-			"code": "' . $js_variationCode . '",
-			"stock": 1,
-			"details": {
-				' . $js_variationDetails . '
-			}
-		}';
+                  			"code": "' . $js_variationCode . '",
+                  			"stock": ' . (Product::getQuantity($product_fields['id_product']) > 0 ? 1 : 0) . ',
```
Comment (bratucornel) [inline ajax.php]: try this: $product->available_now == true ? 1 : 0)

File: ajax.php [hunk 2]
```diff
@@ -236,22 +268,22 @@ function getProductAddToCartJS($id, $vid)
...
             $js_variationDetails = implode(', ', $arr_variationDetails);
             $js_variation = '{
-				"code": "' . $js_variationCode . '",
-				"stock": 1,
-				"details": {
-					' . $js_variationDetails . '
-				}
-			}';
+        				"code": "' . $js_variationCode . '",
+        				"stock": ' . (Product::getQuantity($product_fields['id_product']) > 0 ? 1 : 0) . ',
```
Comment (bratucornel) [inline ajax.php]: $product->available_now == true ? 1 : 0)

File: ajax.php [hunk 3]
```diff
@@ -304,22 +337,22 @@ function getSetVariationJS($id, $vid)
...
```
Comment (bratucornel) [inline ajax.php]: $product->available_now == true ? 1 : 0)

File: retargetingtracker.php [hunk 4]
```diff
@@ -738,11 +730,19 @@ public function hookCreateAccount()
     protected function prepSetEmailJS()
     {
         $customer = $this->context->customer;
+        $birthday = $this->context->customer->birthday;
+        
+        if ($birthday == 'null' || $birthday == '0000-00-00') {
```
Comment (bratucornel) [inline retargetingtracker.php]: check type

File: retargetingtracker.php [hunk 5]
```diff
@@ -764,6 +764,15 @@ public function hookOrderConfirmation($params)
         $discounts = $order->getCartRules();
         $customer = new Customer((int)$order->id_customer);
         $address = new Address((int)$order->id_address_delivery);
+        $birthday = $this->context->customer->birthday;
+        $homePhone = $address->phone;  
+        $mobilePhone = $address->phone_mobile;
+        
+        if ($birthday == 'null' || $birthday == '0000-00-00') {
```
Comment (bratucornel) [inline retargetingtracker.php]: check type

File: retargetingtracker.php [hunk 6]
```diff
@@ -821,7 +841,7 @@ public function hookOrderConfirmation($params)
                 'lastname' => $address->lastname,
                 'firstname' => $address->firstname,
                 'email' => $customer->email,
-                'phone' => ($address->phone == '' ? $address->phone : $address->phone_mobile),
+                'phone' => ($homePhone !== '' ? $homePhone : $mobilePhone),
```
Comment (funnzz) [inline retargetingtracker.php]: Let's fetch the mobile phone number as default. Try this:
"phone": "' . ($mobilePhone !== '' ? $mobilePhone : $homePhone) . '",

**Pull request merged**